### DISCUSSION TITLE
Throw multiple graphql errors

### DISCUSSION THREAD

**Discussion by drmax24** [rebing/graphql-laravel#869] [Q&A]
Is it possible to throw multiple graphql errors from resolver? How to do it? 

Let's say I have a file upload service and I validate file contents. I want to send all errors that are present inside a file as an array to an error section of my graphql response.

**Comment by mfn** [comment]
This library provides the config `error_formatter` and `errors_handler`; the latter which deals with all (multi) errors and latter which is responsible for formatting a singular error.

The defaults for example support `\Illuminate\Validation\ValidationException`, which is a single exception / error but contains a "bag of messages", which usually correlate to fields somehow. What this approach does is using a GraphQL **non-standard** way to have a single entry in the GraphQL `errors` response using the reserved but free-to-use `extensions` key and adds whatever payloads suite it.

This was just an example how it could be done; there two other option quickly coming to my mind:
- throw and handle multiple errors/exception
  Technically it's supported by the underlying graphql-php but I have not tried this approach. It would mean something like `\GraphQL\Executor\ExecutionResult::__construct` and passing multiple `$errors`
- use the GraphQL type system and make the mutation in question return a union, e.g. `mutation myAwesomeUploader : (UploadedFileData | MoreSpecificErrors { error } )`
  i.e. the mutation returns an union consisting of either the object `UploadedFileData` or the object `MoreSpecificErrors` containing multiple errors as strings or whatever you want. You'd have to come up with the types of your own.
  The benefit of this is that you've a typed and flexible system. But it might also be overkill and e.g. the first suggestion with the ValidatinException is more appropriate.

Does that help? Maybe if you describe a bit more details about your use-case I can help more.

**Comment by drmax24** [comment]
>>Technically it's supported by the underlying graphql-php but I have not tried this approach. It would mean something like

Your suggestion is probably a bit more clean and out of the box then:
https://github.com/webonyx/graphql-php/issues/207

That would be interesting to know how to populate errors array exactly. We will try to investigate it...
